Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -0,0 +1,184 @@
+        {
+            for (auto child_link : child_links)
+            {
+                const auto& joint = child_link->parent_joint;
+                AZStd::string joint_name(joint->name.c_str(), joint->name.size());
+                joints[joint_name] = joint;
+                link_visitor(child_link->child_links);
+            }
+        };
+        link_visitor(childLinks);
+        return joints;
+    }
+
+    AZStd::unordered_set<AZStd::string> Utils::GetMeshesFilenames(const urdf::LinkConstSharedPtr& rootLink, bool visual, bool colliders)
+    {
+        AZStd::unordered_set<AZStd::string> filenames;
+        const auto addFilenameFromGeometry = [&filenames](const urdf::GeometrySharedPtr& geometry)
+        {
+            if (geometry->type == urdf::Geometry::MESH)
+            {
+                auto pMesh = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+                if (pMesh)
+                {
+                    filenames.insert(AZStd::string(pMesh->filename.c_str(), pMesh->filename.size()));
+                }
+            }
+        };
+
+        const auto processLink = [&addFilenameFromGeometry, visual, colliders](const urdf::Link& link)
+        {
+            if (visual)
+            {
+                for (auto& p : link.visual_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+            if (colliders)
+            {
+                for (auto& p : link.collision_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+        };
+
+        std::function<void(const std::vector<urdf::LinkConstSharedPtr>&)> linkVisitor =
+            [&](const std::vector<urdf::LinkConstSharedPtr>& child_links) -> void
+        {
+            for (auto link : child_links)
+            {
+                processLink(*link);
+                std::vector<urdf::LinkConstSharedPtr> childVector(link->child_links.size());
+                std::transform(
+                    link->child_links.begin(),
+                    link->child_links.end(),
+                    childVector.begin(),
+                    [](const urdf::LinkSharedPtr& p)
+                    {
+                        return urdf::const_pointer_cast<urdf::Link>(p);
+                    });
+                linkVisitor(childVector);
+            }
+        };
+        linkVisitor({ rootLink });
+        return filenames;
+    }
+
+    /// Finds global path from URDF path
+    AZStd::string Utils::ResolveURDFPath(
+        AZStd::string unresolvedPath, const AZStd::string& urdfFilePath, const AZStd::function<bool(const AZStd::string&)>& fileExists)
+    {
+        if (unresolvedPath.starts_with("package://"))
+        {
+            AZ::StringFunc::Replace(unresolvedPath, "package://", "", true, true);
+            AZ::IO::Path urdfProperPath(urdfFilePath);
+            AZ::IO::Path package_path;
```

Comment:
package_path should be lowerCamelCase per the coding standard.
```suggestion
            AZ::IO::Path packagePath;
```